onnxruntime/core/optimizer/selectors_actions/actions.cc

```diff
@@ -99,8 +94,12 @@ static Status CreateReplacementNode(Graph& graph,
 }
 
 Status ReplaceWithNew::Run(Graph& graph, const NodesToOptimize& selected_nodes) const {
-  const auto op_type = OpType(selected_nodes);
-  ORT_RETURN_IF_ERROR(CreateReplacementNode(graph, selected_nodes, op_type, domain_, value_moves_,
+  const RuntimeState runtime_state{graph, selected_nodes};
+  const auto op_type = OpType(runtime_state);
+  const auto domain = Domain(runtime_state);
+  const auto attributes = Attributes(runtime_state);
+  const auto value_moves = ValueMoves(runtime_state);
+  ORT_RETURN_IF_ERROR(CreateReplacementNode(graph, selected_nodes, op_type, domain, attributes, value_moves,
```

Comment:
Can this usage be made less confusing? ReplaceWithNew has members for these values, but when we need to use them we call virtual methods. 

If something has a member variable I would expect it to be used directly here instead of having an implicit need to call something else.

Reply:
was kind of following the convention set by OpType(). would it be less confusing to remove the data members from ReplaceWithNew?

Reply:
It was slightly different before as the OpType virtual was private so its usage was restricted to the implementation of ReplaceWithNew.

Probably simpler to not have the data members if we having matching virtual methods.